### ISSUE TITLE
hasMany decoding using Row

### ISSUE THREAD

**Issue by zdnk** [groue/GRDB.swift#523]
### What did you do?

Trying to decode joined query.

### What did you expect to happen?

Well, decoded struct with its relations.

### What happened instead?

I am not sure how to approach it without `Decodable`

### Environment

**GRDB flavor(s):** GRDB
**GRDB version:** GRDB-4.0
**Installation method:** CocoaPods
**Xcode version:** 10.2
**Swift version:** 5
**Platform(s) running GRDB:** iOS
**macOS version running Xcode:** latest beta

### Code
```swift
extension Models.Category: FetchableRecord, PersistableRecord {
    
    fileprivate static var tagsAssoc = hasMany(
        CategoryTag.self,
        key: "tags",
        using: CategoryTag.categoryForeignKey
    )
    
    public static var databaseTableName: String {
        return "categories"
    }
    
    static func all() -> QueryInterfaceRequest<Models.Category> {
        return Models.Category.including(required: tagsAssoc)
    }
    
    public init(row: Row) {
        
        self.init(
            id: row["id"],
            name: row["name"],
            position: row["position"],
            parentId: row["parent_id"],
            tags: [], // <===== here I am trying to decode the tags association, but I am not sure how, and I dont want to use `Decodable`
            iconUrl: row["icon_url"]
        )
    }
    
    public func encode(to container: inout PersistenceContainer) {
        container["id"] = id
        container["name"] = name
        container["position"] = position
        container["parent_id"] = parentId
        container["icon_url"] = iconUrl
    }
    
}
```

```swift
fileprivate struct CategoryTag: FetchableRecord, PersistableRecord {
    
    static let categoryForeignKey = ForeignKey(["category_id"])
    
    public static var databaseTableName: String {
        return "category_tags"
    }
    
    let categoryId: Int
    let tag: String
    
    init(categoryId: Int, tag: String) {
        self.categoryId = categoryId
        self.tag = tag
    }
    
    init(row: Row) {
        categoryId = row["category_id"]
        tag = row["tag"]
    }
    
    func encode(to container: inout PersistenceContainer) {
        container["category_id"] = categoryId
        container["tag"] = tag
    }
    
}
```

They query that I wanna decode is the `Models.Category.all()`.

**Comment by groue** [comment]
Hello @zdnk,

Please look at issue #522, which asks the same question.

tl;dr: this feature is not yet there, but is coming soon with GRDB 4. It is actively developed right now.

**Comment by zdnk** [comment]
I see. Would you have any ETA? :)

**Comment by groue** [comment]
No, I don't have any, I am sorry. The main technical difficulties have been solved, that I can say. I mean that the feature will ship, for sure. Meanwhile, please check the other issue for a workaround.

**Comment by groue** [comment]
Draft pull request: #525

**Comment by groue** [comment]
The feature has just landed in GRDB 4.
